File: src/main/java/resonantinduction/CommonProxy.java
/**
 * 
 */
package resonantinduction;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import resonantinduction.core.multimeter.ContainerMultimeter;
import resonantinduction.core.multimeter.PartMultimeter;
import universalelectricity.api.vector.Vector3;
import codechicken.lib.colour.ColourRGBA;
import codechicken.multipart.TMultiPart;
import codechicken.multipart.TileMultipart;
import cpw.mods.fml.common.network.IGuiHandler;

/**
 * @author Calclavia
 * 
 */
public class CommonProxy implements IGuiHandler
{
	public void registerRenderers()
	{

	}

	public void postInit()
	{

	}

	@Override
	public Object getServerGuiElement(int id, EntityPlayer player, World world, int x, int y, int z)
	{
		TileEntity tileEntity = world.getBlockTileEntity(x, y, z);

		if (tileEntity instanceof TileMultipart)
		{
			TMultiPart part = ((TileMultipart) tileEntity).partMap(id);
			if (part instanceof PartMultimeter)
			{
				return new ContainerMultimeter(player.inventory, ((PartMultimeter) part));
			}
		}

		return null;
	}

	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
	{
		return null;
	}

	public boolean isPaused()
	{
		return false;
	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target, float r, float g, float b, boolean split)
	{

	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target, float r, float g, float b)
	{
		this.renderElectricShock(world, start, target, r, g, b, true);
	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target, ColourRGBA color)
	{
		this.renderElectricShock(world, start, target, color.r / 255f, color.g / 255f, color.b / 255f);
	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target, ColourRGBA color, boolean split)
	{
		this.renderElectricShock(world, start, target, color.r / 255f, color.g / 255f, color.b / 255f, split);
	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target)
	{
		this.renderElectricShock(world, start, target, true);
	}

	public void renderElectricShock(World world, Vector3 start, Vector3 target, boolean b)
	{
		this.renderElectricShock(world, start, target, 0.55f, 0.7f, 1f, b);

	}

	public boolean isFancy()
	{
		return false;
	}

}

File: src/main/scala/resonantinduction/atomic/CommonProxy.scala
package resonantinduction.atomic

import cpw.mods.fml.common.network.IGuiHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World
import resonantinduction.atomic.machine.accelerator.{ContainerAccelerator, TileAccelerator}
import resonantinduction.atomic.machine.boiler.{ContainerNuclearBoiler, TileNuclearBoiler}
import resonantinduction.atomic.machine.centrifuge.{ContainerCentrifuge, TileCentrifuge}
import resonantinduction.atomic.machine.extractor.{ContainerChemicalExtractor, TileChemicalExtractor}
import resonantinduction.atomic.machine.quantum.{ContainerQuantumAssembler, TileQuantumAssembler}
import resonantinduction.atomic.machine.reactor.{ContainerReactorCell, TileReactorCell}

class CommonProxy extends IGuiHandler
{
  def preInit
  {
  }

  def init
  {
  }

  def postInit
  {
  }

  def getArmorIndex(armor: String): Int =
  {
    return 0
  }

  def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef =
  {
    return null
  }

  def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef =
  {
    val tileEntity: TileEntity = world.getTileEntity(x, y, z)
    if (tileEntity.isInstanceOf[TileCentrifuge])
    {
      return new ContainerCentrifuge(player.inventory, (tileEntity.asInstanceOf[TileCentrifuge]))
    }
    else if (tileEntity.isInstanceOf[TileChemicalExtractor])
    {
      return new ContainerChemicalExtractor(player.inventory, (tileEntity.asInstanceOf[TileChemicalExtractor]))
    }
    else if (tileEntity.isInstanceOf[TileAccelerator])
    {
      return new ContainerAccelerator(player, (tileEntity.asInstanceOf[TileAccelerator]))
    }
    else if (tileEntity.isInstanceOf[TileQuantumAssembler])
    {
      return new ContainerQuantumAssembler(player.inventory, (tileEntity.asInstanceOf[TileQuantumAssembler]))
    }
    else if (tileEntity.isInstanceOf[TileNuclearBoiler])
    {
      return new ContainerNuclearBoiler(player, (tileEntity.asInstanceOf[TileNuclearBoiler]))
    }
    else if (tileEntity.isInstanceOf[TileReactorCell])
    {
      return new ContainerReactorCell(player, (tileEntity.asInstanceOf[TileReactorCell]))
    }
    return null
  }

  def isFancyGraphics: Boolean =
  {
    return false
  }
}